kensu/utils/helpers.py

```diff
@@ -58,3 +55,18 @@ def extract_ksu_ds_schema(kensu, orig_variable, report=False, register_orig_data
     if register_orig_data:
         kensu.real_schema_df[schema.to_guid()] = orig_variable
     return ds, schema
+
+
+def flatten(d, parent_key='', sep='.'):
+    items = []
+    for k, v in d.items():
+        new_key = parent_key + sep + k if parent_key else k
+        if isinstance(v, dict):
+            items.extend(flatten(v, new_key, sep=sep).items())
+        elif isinstance(v, list):
+            new_key = new_key + '[]'
+            for i in v:
+                items.extend(flatten(i, new_key, sep=sep).items())
```

Comment:
especially in case of list, it might have repeated items? so we might want to remove duplicates or return a `dict(key -> data_type)`